spike_main/spike.cc

```diff
@@ -116,7 +130,7 @@ int main(int argc, char** argv)
   if (mems.empty())
     mems = make_mems("2048");
 
-  sim_t s(isa, nprocs, halted, start_pc, mems, htif_args);
```

Comment:
I wouldn't bother with `std::move` here.  If you just make it a `const vector&` you can elide the copy, but it probably doesn't matter.